server/src/main/java/com/hedera/block/server/producer/ProducerBlockItemObserver.java

```diff
@@ -157,42 +160,47 @@ public ProducerBlockItemObserver(
     public void onNext(
             @NonNull final com.hedera.hapi.block.protoc.PublishStreamRequest publishStreamRequest) {
 
-        try {
-            LOGGER.log(DEBUG, "Received PublishStreamRequest from producer");
-            final BlockItem blockItem =
-                    toPbj(BlockItem.PROTOBUF, publishStreamRequest.getBlockItem().toByteArray());
-            LOGGER.log(DEBUG, "Received block item: " + blockItem);
+        LOGGER.log(DEBUG, "Received PublishStreamRequest from producer");
+        final List<BlockItem> blockItemsPbj = new ArrayList<>();
+        final Counter liveBlockItemsReceived = metricsService.get(LiveBlockItemsReceived);
+        for (final var blockItemProtoc : publishStreamRequest.getBlockItemsList()) {
+            try {
+                final BlockItem blockItem =
+                        toPbj(BlockItem.PROTOBUF, blockItemProtoc.toByteArray());
+                blockItemsPbj.add(blockItem);
+            } catch (ParseException e) {
+                final var errorResponse = buildErrorStreamResponse();
+                publishStreamResponseObserver.onNext(errorResponse);
+                LOGGER.log(
+                        ERROR,
+                        "Error parsing inbound block item from a producer: " + blockItemProtoc,
+                        e);
+
+                // Stop the server
+                serviceStatus.stopWebServer(getClass().getName());
+            }
+            liveBlockItemsReceived.increment();
+        }
 
-            metricsService.get(LiveBlockItemsReceived).increment();
+        // is this log just too much? see below log for proposed...
+        LOGGER.log(DEBUG, "Received block item batch with {} items.", blockItemsPbj.size());
```

Comment:
We _might_ choose to count message tags in the byte array and record that.
We'll need to be able to split the byte array into separate items without parsing anyway, for verification, so that code will be needed regardless.

Reply:
✅ Done!